docs/decisions/20230928-revocation-status-list-expansion-strategy.md

```diff
@@ -0,0 +1,74 @@
+The specification recommends a minimum size of 16 KB for the status list included in a credential.
+However, it does not delineate a maximum size, nor does it provide guidance on how to proceed if the selected status list surpasses its capacity to store information about revoked credentials.
+Put differently, if more credentials are issued than can be accommodated by a 16 KB status list, no specific instructions are provided.
+
+
+## Decision Drivers
+
+We must determine a strategy for expanding the revocation status list to accommodate the increasing number of revoked credentials in the future.
+
+It's crucial to keep in mind that this status list will be part of a "status list credential."
+This credential will be frequently requested through the REST API during verification by verifiers and will be downloaded over the network.
+Therefore, we need to ensure that the status list remains reasonably small in size to prevent any slowdowns in the verification process.
+
+In the future, there might be a need to reorganize the state and possibly merge various status lists into a single comprehensive registry for verifiers to depend on. This is not the current scenario, as each Prism agent currently maintains status lists specific to their respective tenants.
+
+Absolutely, it's crucial to avoid overengineering the solution. This ensures that the code remains manageable and easy to maintain in the long run.
+
+
+## Considered Options
+
+
+Option 1: Increment status list size as we approach its limit:
+
+We'll enhance the status list by simply doubling its size.
+If we opt for the minimum size of 16 KB for the revocation status list, it can manage up to 131,072 revocable credentials before requiring expansion.
+It's important to note that this number represents the potential capacity for revocable credentials to be issued, not necessarily the actual number of credentials that have been revoked.
+Even unrevoked credentials still occupy space.
+
+Option 2: Generate multiple revocation status lists as the previous one reaches its limit
+
+With this approach, we'll generate and store multiple status list credentials.
+It will be crucial to ensure that each credential is linked to a specific status list, allowing us to track where the revocation information is stored.
+If we stick with the smallest recommended status list size, one revocation status list can hold information about 131,072 revocable credentials.
+
+
+## Decision Outcome
+
+
+### Positive Consequences
+
+
+### Negative Consequences
+
+
+## Pros and Cons of the Options
+
+Option 1 offers the primary advantage of being straightforward to implement.
+Additionally, if we ever need to merge status lists in the future, it would likely be a more seamless process.
+However, it's important to note that Option 2 isn't significantly more challenging to implement, so we shouldn't overly prioritize this consideration.
```

Comment:
If implementation complexity is not an issue, then I would consider going for option 2. The thing is, with AnonCreds, there is no way to extend the status list size. The capacity (maximum number of credentials) is defined once, at the time the revocation registry definition is created, and it can't be changed afterwards.

We could be tempted to push back option 2 for AnonCreds as well, and say we will create an initial status list with a capacity of e.g. 1M credentials, but this won't be efficient at all. I've investigated AnonCreds revocation lately, and it appears that the size of the attached TAILS FILE is growing rapidly and proportionally to the capacity (e.g. 8.4MB for 32.768 VCs!). And the file has to be resolved/downloaded by the holder at some point during the issuance flow.

I know that AnonCreds revocation is uncommitted for PI4, but it will come just after. So we will need option 2 in the near future, anyway.

Reply:
Yes, AnonCreds has a similar approach. It uses a status list (published on VDR) that contains a revocation array (`0`or `1`) with each item/index assigned to a specific VC. The format is different, but the principles/mechanisms are the same. However, extending the array afterwards `IS NOT` an option with AnonCreds.
So, I mean that implementing Option 2 and handling multiple status list, would be a step in the right direction as part of the logic will probably be reusable for AnonCreds.